lib/wibox/widget/piechart.lua

```diff
@@ -0,0 +1,212 @@
+---------------------------------------------------------------------------
+-- Display percentage in a circle.
+--
+-- Note that this widget makes no attempts to prevent overlapping labels or
+-- labels drawn outside of the widget boundaries.
+--
+--@DOC_wibox_widget_defaults_piechart_EXAMPLE@
+-- @author Emmanuel Lepage Valle
+-- @copyright 2012 Emmanuel Lepage Vallee
+-- @release @AWESOME_VERSION@
+-- @classmod wibox.widget.piechart
+---------------------------------------------------------------------------
+
+local color     = require( "gears.color"       )
+local base      = require( "wibox.widget.base" )
+local beautiful = require( "beautiful"         )
+local util      = require( "awful.util"        )
+local pie       = require( "gears.shape"       ).pie
+local unpack    = unpack or table.unpack -- luacheck: globals unpack (compatibility with Lua 5.1)
+
+local module = {}
+
+local piechart = {}
+
+local function draw_label(cr,angle,radius,center_x,center_y,text)
+    cr:move_to(center_x+(radius/2)*math.cos(angle),center_y+(radius/2)*math.sin(angle))
+    cr:line_to(center_x+(1.5*radius)*math.cos(angle),center_y+(1.5*radius)*math.sin(angle))
```

Comment:
Perhaps `rel_line_to` helps here? The above `line_to` can be replaced with (at least I think so):
```lua
cr:rel_line_to(radius*math.cos(angle), radius*math.sin(angle))
```